src/webgpu/api/validation/copyTextureToTexture.spec.ts

```diff
@@ -0,0 +1,265 @@
+export const description = `
+copyTextureToTexture tests.
+
+Test Plan: (TODO(jiawei.shao@intel.com): add all the remaining tests)
+* the source and destination texture
+  - the {source, destination} texture is {invalid, valid}.
+  - mipLevel {>, =, <} the mipmap level count of the {source, destination} texture.
+  - the source texture is created {with, without} GPUTextureUsage::CopySrc.
+  - the destination texture is created {with, without} GPUTextureUsage::CopyDst.
+* sample count
+  - the sample count of the source texture {is, isn't} equal to the one of the destination texture
+  - when the sample count is greater than 1:
+    - it {is, isn't} a copy of the whole subresource of the source texture.
+    - it {is, isn't} a copy of the whole subresource of the destination texture.
+`;
+
+import { poptions, params } from '../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+import { kTextureUsages } from '../../capability_info.js';
+
+import { ValidationTest } from './validation_test.js';
+
+class F extends ValidationTest {
+  TestCopyTextureToTexture(
+    source: GPUTextureCopyView,
+    destination: GPUTextureCopyView,
+    copySize: GPUExtent3D,
+    isSuccess: boolean
+  ): void {
+    const commandEncoder = this.device.createCommandEncoder();
+    commandEncoder.copyTextureToTexture(source, destination, copySize);
+
+    this.expectValidationError(() => {
+      commandEncoder.finish();
+    }, !isSuccess);
+  }
+}
+
+export const g = makeTestGroup(F);
+
+g.test('copy_with_invalid_texture').fn(async t => {
+  const validTexture = t.device.createTexture({
+    size: { width: 4, height: 4, depth: 1 },
+    format: 'rgba8unorm',
+    usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.COPY_DST,
+  });
+
+  const errorTexture = t.getErrorTexture();
+
+  t.TestCopyTextureToTexture(
+    { texture: errorTexture },
+    { texture: validTexture },
+    { width: 1, height: 1, depth: 1 },
+    false
+  );
+  t.TestCopyTextureToTexture(
+    { texture: validTexture },
+    { texture: errorTexture },
+    { width: 1, height: 1, depth: 1 },
+    false
+  );
+});
+
+g.test('mipmap_level')
+  .params([
+    {
+      srcMipLevelCount: 1,
```

Comment:
uNit: Consider using shorter names here so these can fit on one line each (would be easier to read). Maybe change `MipLevel` to just `Level`.

Reply:
Done